From 0df496515045f5af9302718694ab5417d6fe902c Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Wed, 22 Nov 2017 10:18:37 +0800
Subject: [PATCH 2136/2241] arm64: dts: marvell: add description for
 armada-37xx SATA LED blinking

Add SATA LED blink policy constants in new file "leds-ahci-mvebu.h";
Add description for armada-37xx SATA LED blinking:
- The LED controlled by GPIO 11 of GPIO north bank is used for SATA
  ACT/PRESENT LED.;
- The SATA LED blink policy is: the LED is on when idle, and blinks
  off every 10ms when activity is detected.

Change-Id: I7c0f1b7cf488adb3e519adadf5a14ba4905cb066
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/47186
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts |  2 ++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi   |  1 +
 include/dt-bindings/leds/leds-ahci-mvebu.h     | 27 ++++++++++++++++++++++++++
 3 files changed, 30 insertions(+)
 create mode 100644 include/dt-bindings/leds/leds-ahci-mvebu.h

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 409c42d..f9baabc 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -226,6 +226,8 @@
 			sata@e0000 {
 				target-supply = <&reg_5v_sata>;
 				phys = <&a3700_comphy 2 COMPHY_SATA0>;
+				marvell,led-gpio = <&gpionb 11 GPIO_ACTIVE_LOW>;
+				marvell,blink-policy = <LEDS_AHCI_MVEBU_BLINK_OFF>;
 				status = "okay";
 			};
 
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index e6f170a..14c600c 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -48,6 +48,7 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/phy/phy-utmi-mvebu.h>
 #include <dt-bindings/phy/phy-comphy-mvebu.h>
+#include <dt-bindings/leds/leds-ahci-mvebu.h>
 
 / {
 	model = "Marvell Armada 37xx SoC";
diff --git a/include/dt-bindings/leds/leds-ahci-mvebu.h b/include/dt-bindings/leds/leds-ahci-mvebu.h
new file mode 100644
index 0000000..fba5c41
--- /dev/null
+++ b/include/dt-bindings/leds/leds-ahci-mvebu.h
@@ -0,0 +1,27 @@
+/*
+ * This header provides constants for AHCI MVEBU LED bindings.
+ *
+ * Copyright (C) 2017 Marvell
+ *
+ * Ken Ma <make@marvell.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#ifndef _DT_BINDINGS_LEDS_AHCI_MVEBU_H
+#define _DT_BINDINGS_LEDS_AHCI_MVEBU_H
+
+/*
+ * 0    LEDS_AHCI_MVEBU_OFF - the LED is not activated on activity
+ * 1    LEDS_AHCI_MVEBU_BLINK_ON - the LED blinks on every 10ms when activity
+ *                                 is detected.
+ * 2    LEDS_AHCI_MVEBU_BLINK_OFF - the LED is on when idle, and blinks off
+ *                                  every 10ms when activity is detected.
+ */
+#define LEDS_AHCI_MVEBU_OFF			0
+#define LEDS_AHCI_MVEBU_BLINK_ON		1
+#define LEDS_AHCI_MVEBU_BLINK_OFF		2
+
+#endif /* _DT_BINDINGS_LEDS_AHCI_MVEBU_H */
-- 
2.7.4

